halo2_proofs/src/plonk/circuit.rs

```diff
@@ -291,6 +337,25 @@ impl TableColumn {
     }
 }
 
+/// A challenge squeezed from transcript after advice columns at the phase have been committed.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct Challenge {
+    index: usize,
+    phase: Phase,
+}
+
+impl Challenge {
+    /// The index of this challenge.
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    /// The phase where this challenge will be squeezed in the end.
+    pub fn phase(&self) -> Phase {
```

Comment:
This should not be public:
```suggestion
    pub(crate) fn phase(&self) -> Phase {
```

Reply:
Updated in [`902965c`](https://github.com/zcash/halo2/pull/593/commits/902965c385833ada61def929905c7431ff36e78b).